redfish-core/lib/metric_definition.hpp

```diff
@@ -0,0 +1,404 @@
+                                 boost::system::errc::success),
+                             counter->first);
+                }
+            };
+
+        for (const std::string& chassisName : chassisNames)
+        {
+            ++counter->second;
+            retrieveUriToDbusMap(chassisName,
+                                 stringViewToStringRef(sensors::sensorsNodeStr),
+                                 handleRetrieveUriToDbusMap);
+        }
+    });
+}
+
+} // namespace telemetry
+
+inline void requestRoutesMetricDefinitionCollection(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/TelemetryService/MetricDefinitions/")
+        .privileges(privileges::getMetricDefinitionCollection)
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+                telemetry::mapRedfishUriToDbusPath(
+                    [asyncResp](boost::system::error_code ec,
+                                const boost::container::flat_map<
+                                    std::string, std::string>& uriToDbus) {
+                        if (ec)
+                        {
+                            messages::internalError(asyncResp->res);
+                            BMCWEB_LOG_ERROR(
+                                "mapRedfishUriToDbusPath error: {}",
+                                ec.value());
+                            return;
+                        }
+
+                        std::set<std::string> members;
+
+                        for (const auto& [uri, dbusPath] : uriToDbus)
+                        {
+                            members.insert(
+                                telemetry::mapSensorToMetricDefinition(
+                                    dbusPath));
+                        }
+
+                        for (const std::string& odataId : members)
+                        {
+                            asyncResp->res.jsonValue["Members"].push_back(
+                                {{"@odata.id", odataId}});
+                        }
+
+                        asyncResp->res.jsonValue["Members@odata.count"] =
+                            asyncResp->res.jsonValue["Members"].size();
+                    });
+
+                asyncResp->res.jsonValue["@odata.type"] =
+                    "#MetricDefinitionCollection.MetricDefinitionCollection";
+                asyncResp->res.jsonValue["@odata.id"] =
+                    "/redfish/v1/TelemetryService/MetricDefinitions";
+                asyncResp->res.jsonValue["Name"] =
+                    "Metric Definition Collection";
+                asyncResp->res.jsonValue["Members"] = nlohmann::json::array();
+                asyncResp->res.jsonValue["Members@odata.count"] = 0;
+            });
+}
+
+inline void requestRoutesMetricDefinition(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/TelemetryService/MetricDefinitions/<str>/")
```

Comment:
The new way to do this is these ROUTEs sit below and then a function for each route 
e.g. https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/power_supply.hpp#L594

It would make this more readable